clang/test/CIR/Lowering/ThroughMLIR/while.cir

```diff
@@ -0,0 +1,48 @@
+// RUN: cir-opt %s -cir-to-mlir -o %t.mlir
+// RUN: FileCheck %s --input-file %t.mlir
+
+!s32i = !cir.int<s, 32>
+module {
```

Comment:
Please either add a comment with the C/C++ code we could use to regen this test in case CIR changes significantly or add a test coming from C/C++ source.

Reply:
Ok, I will add some comment and test case

Reply:
Hello, I add a test coming from C source , can you review it?